Source/Core/Core/HLE/HLE_VarArgs.h

```diff
@@ -0,0 +1,190 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Common/Align.h"
+#include "Common/CommonTypes.h"
+
+#include "Core/HW/Memmap.h"
+#include "Core/PowerPC/PowerPC.h"
+
+#include <type_traits>
+
+namespace HLE
+{
+namespace SystemVABI
+{
+// SFINAE
+template <typename T>
+struct IsArgPointer
+    : std::integral_constant<bool, std::is_union<T>::value || std::is_class<T>::value>
```

Comment:
`std::is_union<T>::value` => `std::is_union<T>()`? (not sure if this is possible here, feel free to ignore if it's not)